routers/api/v1/repo/issue.go

```diff
@@ -189,9 +189,18 @@ func SearchIssues(ctx *context.APIContext) {
 	if strings.IndexByte(keyword, 0) >= 0 {
 		keyword = ""
 	}
+	// this api is also used in UI,
+	// so the default limit is set to fit UI needs
+	limit := ctx.FormInt("limit")
+	if limit == 0 {
+		limit = setting.UI.IssuePagingNum
+	} else if limit > setting.API.MaxResponseItems {
+		limit = setting.API.MaxResponseItems
+	}
```

Comment:
:raised_eyebrow:
Sounds like a bug report waiting to happen.

Reply:
These lines just moved from line 222 - 230